### ISSUE TITLE
SX1101 incorrectly reported for parameter that hides field

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1707]
SX1101 is reported for the use of `this.` in the following code. The code fix results in a compilation warning (assignment made to same variable) and (more importantly) a change in code behavior.

```csharp
class ClassName
{
  int items;
  void Method(int items)
  {
    this.items = items;
  }
}
```

**Comment by vweijsters** [comment]
Grabbing this

**Comment by vweijsters** [comment]
@sharwell How did you get this code to produce an SX1101 diagnostic?
This case is covered by the `VerifyThatNecessaryPrefixWillNotProduceDiagnosticsAsync` test case and works fine for me.

**Comment by sharwell** [comment]
1. I enabled SX1101 and disabled SA1101 in StyleCop.Analyzers.ruleset.
2. I ran fix all in solution for SX1101.
3. I reverted just the change to StyleCop.Analyzers.ruleset.
4. I ran fix all in solution for SA1101.

After the above steps, you can diff the current code against HEAD to see the places where the SX1101 code fix made a mistake.

This bug report was a simplification of one group of issues I saw in the result.

**Comment by vweijsters** [comment]
OK, that is clear. Unfortunately for the description you picked the only example that is already handled :wink:

**Comment by sharwell** [comment]
Just my luck! :smile:
